hwy/ops/ppc_vsx-inl.h

```diff
@@ -3771,67 +3771,98 @@ HWY_API size_t CountTrue(D /* tag */, MFromD<D> mask) {
   return PopCount(detail::BitsFromMask(mask));
 }
 
+#if HWY_PPC_HAVE_9 && (!HWY_PPC_HAVE_10 || HWY_IS_BIG_ENDIAN)
+namespace detail {
+
+template <class V>
+static HWY_INLINE size_t VsxCntlzLsbb(V v) {
+#if HWY_COMPILER_GCC_ACTUAL && HWY_COMPILER_GCC_ACTUAL < 1200 && \
+    HWY_IS_LITTLE_ENDIAN
+  // Use inline assembly to work around bug in GCC 11 and earlier on
+  // little-endian PPC9
+  int idx;
+  __asm__("vctzlsbb %0,%1" : "=r"(idx) : "v"(v.raw));
```

Comment:
The vec_cntlz_lsbb intrinsic is supposed to compile down to a vclzlsbb instruction on big-endian PPC9 but compile down to a vctzlsbb instruction on little-endian PPC9.

GCC 11 and earlier have a bug where the vec_cntlz_lsbb is compiled down to vctzlsbb instead of vclzlsbb.

The Power Vector Intrinsic Programming Reference also states that the vec_cntlz_lsbb intrinisic should be compiled down to a vclzlsbb instruction on little-endian PPC9 and a vctzlsbb instruction on big-endian PPC9.

The vec_cntlz_lsbb intrinsic is supposed to return the number of leading zero bytes on both little-endian PPC9 and big-endian PPC9.

Note that the hardware lane order on little-endian Altivec/VSX is opposite that of big-endian Altivec/VSX, but the GCC/Clang language extensions and most of the Altivec/VSX intrinsics are aware of the lane order difference.

I have also compiled and run find_test and mask_test with the fixes in this pull request for little-endian PPC9 with GCC 11, GCC 12, and Clang 16, and both find_test and mask_test pass on little-endian PPC9 when compiled with GCC 11, GCC 12, or Clang 16 with the fixes in this pull request.

Here is a link to a snippet that shows the bug with the vec_cntlz_lsbb and vec_cnttz_lsbb intrinsics on little-endian PPC9 when compiled with GCC 11 or earlier (with GCC 13 and Clang generate correct code): https://godbolt.org/z/YzK55haEY

Reply:
Oh, I see, thanks for clarifying. The intrinsics are being 'clever' by swapping L/T under the hood. That's dangerous :)
Your comments are valuable here and in future I'd also advise putting some small mention into the source code as well, because future code readers might check the revision history but not necessarily see the discussion here.